pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -989,26 +1163,62 @@ def __init__(
         if not control_values:
             control_values = "1" * len(control_wires)
 
-        control_int = self._parse_control_values(control_wires, control_values)
         self.control_values = control_values
-
-        self._padding_left = control_int * 2
-        self._padding_right = 2 ** len(wires) - 2 - self._padding_left
-        self._CX = None
+        self.control_int = self._parse_control_values(self.control_wires, self.control_values)
+        self._cx = None
 
         super().__init__(*params, wires=wires, do_queue=do_queue)
 
     @property
     def num_params(self):
         return 0
 
-    def _matrix(self, *params):
-        if self._CX is None:
-            self._CX = block_diag(
-                np.eye(self._padding_left), PauliX.matrix, np.eye(self._padding_right)
-            )
+    @staticmethod
+    def compute_matrix(control_int, n_wires):
```

Comment:
This is a case where we could do it many ways...we could pass `params, hyperparams` and do the entire computation in this method (only). Or we compute things like `control_int` in the init method and add it to the `hyperparams` (ignoring others). Here I overwrite the signature to take very specific inputs, and overwrite `matrix` to provide them.